CHANGES.md

```diff
@@ -1,6 +1,473 @@
+  (see 'Methods on All Domain Types' in the 'Domains' chapter of the spec)
+* added support for using GASNet's `smp` conduit
+
+Feature Improvements
+--------------------
+* dramatically improved support for user-defined initializers
+  (see https://chapel-lang.org/docs/1.17/technotes/initializers.html)
+* made `real(32)` variables significantly more usable
+  (see 'Implicit Numeric, Bool, and Enumeration Conversions' in the spec)
+* made string->value casts throw rather than halt() when encountering errors
+* `Owned(c)` and `Shared(c)` now coerce to type `c`
+  (see https://chapel-lang.org/docs/1.17/modules/packages/OwnedObject.html)
+* `Owned(d)`/`Shared(d)` now coerce to `Owned(c)`/`Shared(c)` for `class d: c`
+  (see https://chapel-lang.org/docs/1.17/modules/packages/OwnedObject.html)
+* writing out an `Owned(c)` now simply prints the `c` object
+* improved support for casting arrays to strings
+* made clear() for an array of records call the records' deinitializers
+* added support for defining multiple config types in a single statement
+  (see 'Type Aliases' in the 'Types' chapter of the language specification)
+* made error-handling propagate through forwarded methods
+* added support for invoking recursive parallel iterators via forall loops
+* enabled wide pointers to be cast to `c_void_ptr`
+* improved support for enums with non-trivial initialization expressions
+* improved default argument handling
+
+Removed Features
+----------------
+* removed the `msg` field from the `Error` class
+* removed deprecated `Barrer` module in favor of `Barriers`
+* removed deprecated `LayoutCSR` module in favor of `LayoutCS`
+* removed `ReplicatedDist` in favor of `Replicated`
+
+Standard Modules/Library
+------------------------
+* added Bessel functions to `Math` module
+  (see https://chapel-lang.org/docs/1.17/modules/standard/Math.html#Math.j0)
+* added support for joinPath(), isAbsPath(), commonPath() to the `Path` module
+  (see https://chapel-lang.org/docs/1.17/modules/standard/Path.html)
+* added support for channel.advancePastByte() to read until a particular byte
+  (see https://chapel-lang.org/docs/1.17/modules/standard/IO.html#IO.channel.advancePastByte)
+* added versions of channel.mark(), commit(), revert() when locking==false
+  (see https://chapel-lang.org/docs/1.17/modules/standard/IO.html#IO.channel.mark)
+* improved support for I/O on `date`, `time`, and `datetime` types in `DateTime`
+  (see https://chapel-lang.org/docs/1.17/modules/standard/DateTime.html#DateTime.date.readWriteThis)
+* made DateTime.time.tzinfo `Shared` for automated memory management
+  (see https://chapel-lang.org/docs/1.17/modules/standard/DateTime.html#DateTime.time.init)
+* array `push*` methods now use `in` intents to improve their use with Owned
+  (see https://chapel-lang.org/docs/1.17/builtins/internal/ChapelArray.html#ChapelArray.push_back)
+* added support for using PCG random streams' iterate() in a zippered forall
```

Comment:
Actually this worked in 1.15 but not in 1.16. I think it should be listed as a bug fix.